microsoft-edge/extensions-chromium/developer-guide/native-messaging.md

```diff
@@ -58,79 +56,165 @@ The host manifest file must be a valid json file containing the following fields
     }
 ```  
 
+The host manifest file must be a valid JSON file that contains the following keys.  
+
+:::row:::
+   :::column span="1":::
+      `name`  
+   :::column-end:::
+   :::column span="2":::
+      Specifies the name of the native messaging host.  Clients pass this string to `runtime.connectNative` or `runtime.sendNativeMessage`.  
+      
+      *   This value must only contain lowercase alphanumeric characters, underscores, and dots.  
+      *   This value must not start or end with a dot, and a dot must not be followed by another dot.  
+   :::column-end:::
+:::row-end:::  
+:::row:::
+   :::column span="1":::
+      `description`  
+   :::column-end:::
+   :::column span="2":::
+      Describes the application.  
+   :::column-end:::
+:::row-end:::  
+:::row:::
+   :::column span="1":::
+      `path`  
+   :::column-end:::
+   :::column span="2":::
+      Specifies the path to the native messaging host binary.  
+      
+      *   On Windows devices, you may use relative paths to the directory that contains the manifest file.  
+      *   On macOS and Linux, the path must be absolute.  
+      
+      The host process starts with the current directory set to the directory that contains the host binary.  For example \(Windows\), if this parameter is set to `C:\Application\nm_host.exe`, the binary is started using the current directory \(`C:\Application\`\).  
+   :::column-end:::
+:::row-end:::  
+:::row:::
+   :::column span="1":::
+      `type`  
+   :::column-end:::
+   :::column span="2":::
+      Specifies the type of the interface used to communicate with the native messaging host.  This value instructs Microsoft Edge to use `stdin` and `stdout` to communicate with the host.  
+      The only acceptable value is `stdio`.  
+   :::column-end:::
+:::row-end:::  
+:::row:::
+   :::column span="1":::
+      `allowed_origins` 
+   :::column-end:::
+   :::column span="2":::
+      Specifies the list of extensions that have access to the native messaging host.  To enable your application to identify and communicate with an extension, in your native messaging host manifest file set the following value.  
+      
+      ```json
+      "allowed_origins": ["chrome-extension://{microsoft_catalog_extension_id}"]
+      ```  
+   :::column-end:::
+:::row-end:::  
+
+Sideload your extension to test native messaging with the host.  
+To sideload of your extension during development, complete the following steps.  
```

Comment:
To sideload of your extension => To sideload your extension

Reply:
To sideload your extension during development and retrieve microsoft_catalog_extension_id, complete the following steps.

Reply:
Updated.

```md
Sideload your extension to test native messaging with the host.  
To sideload your extension during development and retrieve `microsoft_catalog_extension_id`, complete the following steps.  
```